dashboard/modules/job/job_head.py

```diff
@@ -128,7 +132,66 @@ class JobHead(dashboard_utils.DashboardHeadModule):
     def __init__(self, dashboard_head):
         super().__init__(dashboard_head)
         self._dashboard_head = dashboard_head
-        self._job_manager = None
+        self._job_info_client = None
+        # this is a queue of JobAgentSubmissionClient
+        self._agents = OrderedDict()
+
+        self._agents_pool = dict()
+
+    async def choose_agent(self) -> Optional[JobAgentSubmissionClient]:
+        """
+        Try to disperse as much as possible to select one of
+        the `CANDIDATE_AGENT_NUMBER` agents to solve requests.
+        the agents will not pop from `self._agents` unless
+        it's dead.
+        Follow the steps below to select the agent client:
+            1. delete dead agent from `self._agents`, make sure
+               the `JobAgentSubmissionClient` in `self._agents`
+               is always available.
+            2. Attempt to put new agents into `self._agents` until
+               its size is `CANDIDATE_AGENT_NUMBER`
+            3. Returns the element at the head of the `self._agents`
+               and put it into `self._agents` again.
+        """
+        # the number of agents which has an available HTTP port.
+        while True:
+            raw_agent_infos = await DataOrganizer.get_all_agent_infos()
+            agent_infos = {
+                key: value
+                for key, value in raw_agent_infos.items()
+                if value.get("httpPort", -1) > 0
+            }
+            if len(agent_infos) > 0:
+                break
+            await asyncio.sleep(dashboard_consts.WAIT_RAYLET_START_INTERVAL_SECONDS)
```

Comment:
This enum is very wired. Change the name to better reflect its usage.

Reply:
how about `TRY_TO_GET_AGENT_INFO_INTERVAL_SECONDS`?